core/open.py

```diff
@@ -0,0 +1,118 @@
+
+
+class OpenTask:
+    def __init__(self, args):
+        self.args = args
+        self.config = RuntimeConfig.from_args(args)
+        self.model_path = self.config.source_paths[0]
+        self.target_path = self.config.target_path
+        self.manifest = self._get_manifest()
+        self.open_command = self._get_open_command()
+
+    def _get_open_command(self):
+        open_command = os.environ.get(
+            "DBT_HELPER_OPEN_COMMAND", os.environ.get("EDITOR", DEFAULT_OPEN_COMMAND)
+        )
+        return open_command
+
+    def _get_manifest(self):
+        """
+        This subcommand uses the manifest file, whereas other subcommands import
+        the manifest object from dbt (which requires the project to be parsed).
+        Using the manifest file is significantly faster, so is preferred in this
+        case.
+        """
+        manifest_path = os.path.join(self.target_path, MANIFEST_FILE)
+        try:
+            with open(manifest_path) as f:
+                manifest = json.load(f)
+            return manifest
+        except FileNotFoundError as err:
+            raise Exception(
+                "Could not find {} file. You may need to run dbt compile first".format(
+                    MANIFEST_FILE
+                )
+            )
+
+    def _get_model_file_names(self):
+        """ Return a dictionary of the form:
+        {
+            'my_model_1': {
+                'source': file_path,
+                'compiled': file_path,
+                'run': file_path
+            },
+            'my_model_2': {..}
+        }
+        """
+
+        file_names = {}
+        for node in self.manifest["nodes"].values():
+            if node["resource_type"] == "model":
+                model_name = node["name"]
+                relative_path = node["path"]
+                original_file_path = node["original_file_path"]
+                package_name = node["package_name"]
+
+                root_path = [node["root_path"], original_file_path]
+
+                compiled_path = [
+                    self.target_path,
+                    COMPILED_DIR,
+                    package_name,
+                    relative_path,
+                ]
+
+                run_path = [self.target_path, RUN_DIR, package_name, relative_path]
+
+            file_names[model_name] = {
+                "source": root_path,
+                "compiled": compiled_path,
+                "run": run_path,
+            }
+
+        return file_names
+
+    def run(self):
+
+        model_file_names = self._get_model_file_names()
+
+        model_files = model_file_names.get(self.args.model_name)
+
+        if model_files is None:
+            raise Exception(
+                "Could not find a model '{}'. You may need to run dbt compile first.".format(
+                    self.args.model_name
+                )
+            )
+
+        file_to_open = model_files.get(self.args.code_type)
+        file_path = os.path.join(*file_to_open)
```

Comment:
For some reason I feel like doing this makes more sense in the `_get_model_file_names` method just since it makes the interface more obvious (the `model_file_names` dict is actually a mapping between paths and objects).